src/DocTests.jl

```diff
@@ -124,7 +124,11 @@ function doctest(ctx::DocTestContext, block_immutable::Markdown2.CodeBlock)
     if startswith(lang, "jldoctest")
         # Define new module or reuse an old one from this page if we have a named doctest.
         name = match(r"jldoctest[ ]?(.*)$", split(lang, ';', limit = 2)[1])[1]
-        sym = isempty(name) ? gensym("doctest-") : Symbol("doctest-", name)
+        if isempty(name)
+            sym = Symbol("__doctest__", filter(!=('#'), string(gensym())))
```

Comment:
It looks like `!=('#')` doesn't work on 1.0, but this should be fine (and, arguably, a bit more readable):

```suggestion
            sym = Symbol("__doctest__", lstrip(string(gensym()), '#'))
```